### PR TITLE
Bump version to 2.4.0.0

### PR DESCRIPTION
**Pull request by heemin32**
Signed-off-by: Heemin Kim <heemin@amazon.com>

### Description
Bump version to 2.4.0.0

### Issues Resolved
https://github.com/opensearch-project/dashboards-maps/issues/69

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by junqiu-lei** [comment]
I think the [opensearch_dashboards.json](https://github.com/opensearch-project/dashboards-maps/blob/b00906788081fec28b183eb95b9368cc87235c16/src/plugins/custom_import_map/opensearch_dashboards.json) also need a version bump.

**Comment by heemin32** [comment]
> I think there are other places also need version bump.
> 
> https://github.com/opensearch-project/dashboards-maps/blob/b00906788081fec28b183eb95b9368cc87235c16/.github/workflows/unit-tests-workflow-for-customImportMap.yml#L11
> 
> 
> https://github.com/opensearch-project/dashboards-maps/blob/b00906788081fec28b183eb95b9368cc87235c16/.github/workflows/cypress-workflow.yml#L12-L13
> 
> 
> https://github.com/opensearch-project/dashboards-maps/blob/b00906788081fec28b183eb95b9368cc87235c16/src/plugins/custom_import_map/opensearch_dashboards.json#L3-L4

I could not update unit-tests-workflow-for-customImportMap.yml as it fails unit test when I bump the version. Let me create an issue to follow up.

I could not update cypress-workflow.yml as 2.4 snapshot is not ready for OS and OSD yet. Following in https://github.com/opensearch-project/dashboards-maps/issues/71

File: src/plugins/custom_import_map/opensearch_dashboards.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "id": "customImportMapDashboards",
-    "version": "2.3.0.0",
-    "opensearchDashboardsVersion": "2.3.0",
+    "version": "2.4.0.0",
+    "opensearchDashboardsVersion": "2.3.1",
```
Comment (navneet1v) [inline src/plugins/custom_import_map/opensearch_dashboards.json:4]: this is interesting, it should also be 2.4.0

Comment (heemin32) [reply src/plugins/custom_import_map/opensearch_dashboards.json]: OpenSearchDashboards does not have 2.4 branch yet. So, cannot set it as 2.4.0 for now.

**Review by junqiu-lei — APPROVED** [review]
LGTM, thanks.

**Review by naveentatikonda — APPROVED** [review]

**Pull request merged**